feat: New landing page design

pranshuchittora opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
New landing page design

### Fixed Issues
<!-- Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing -->
Closes #2938 
### Tests
<!--- 
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->

### QA Steps
<!--- 
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image 
--->
- Match the UI with the Figma Designs
- Previously we are only showing Terms in the mobile. In the new design we are showing Terms as well as license.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

![Screenshot from 2021-07-08 02-18-44](https://user-images.githubusercontent.com/32242596/124827072-39552500-df93-11eb-88cf-37e1bd91d6b0.png)


#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

![Screenshot from 2021-06-18 00-16-10](https://user-images.githubusercontent.com/32242596/122455849-7cdaf580-cfca-11eb-8c9e-5ed6c6930cef.png)

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

<img width="1205" alt="Screenshot 2021-07-08 at 2 24 23 AM" src="https://user-images.githubusercontent.com/32242596/124827505-bbdde480-df93-11eb-9fa7-18cf511cdf5f.png">

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

![Simulator Screen Shot - iPhone 12 - 2021-06-23 at 21 47 43](https://user-images.githubusercontent.com/32242596/123132827-c749ef80-d46c-11eb-92bb-1a1271917340.png)


#### Android
<!-- Insert screenshots of your changes on the Android platform-->

![Screenshot from 2021-06-18 00-18-11](https://user-images.githubusercontent.com/32242596/122456043-b3187500-cfca-11eb-87e5-d61ab844a765.png)

shawnborton commented:
Looks like the mobile view should have more padding around the inner logo/UI/etc.

shawnborton commented on src/styles/variables.js:
```diff
@@ -14,9 +14,11 @@ export default {
     fontSizeLabel: 13,
     fontSizeNormal: 15,
     fontSizeLarge: 17,
+    fontSizeExtraLarge: 36,
```
I think a more accurate description for this might be `fontSizeHero` in that it's the large font size we might use for big hero moments.

pranshuchittora replied:
Done

shawnborton commented on src/styles/variables.js:
```diff
@@ -14,9 +14,11 @@ export default {
     fontSizeh1: 19,
     iconSizeExtraSmall: 12,
     fontSizeNormalHeight: 20,
+    lineHeightExtraLarge: 40,
```
Same comment as above, perhaps `lineHeightHero` makes more sense.

pranshuchittora replied:
Done

Jag96 commented on src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js:
```diff
@@ -1,97 +1,59 @@
-    shouldShowWelcomeText: PropTypes.bool,
-    shouldShowWelcomeScreenshot: PropTypes.bool,
+    // shouldShowWelcomeText: PropTypes.bool,
+    // shouldShowWelcomeScreenshot: PropTypes.bool,
```
Is there a reason these are commented out? Please either remove them if they are not needed, or uncomment them. Same for the other instances in this diff.

pranshuchittora replied:
I am some queries to be resolved 
https://github.com/Expensify/Expensify.cash/issues/2938#issuecomment-847191266

trjExpensify replied:
👋 @pranshuchittora were your queries resolved here? Please let us know if not.

pranshuchittora replied:
Changes done.
I have removed the props :)

Jag96 commented on src/pages/signin/SignInPageLayout/SignInPageLayoutNarrow.js:
```diff
@@ -1,97 +1,59 @@
     </ScrollView>
 );
 
+
```
Unnecessary extra space

pranshuchittora replied:
Done

Jag96 commented on src/styles/utilities/spacing.js:
```diff
@@ -195,6 +199,26 @@ export default {
+        paddingHorizontal: 24,
+    },
+
+    ph7: {
```
Is this used? If not, let's remove it and any other unused classes that have been added here. For example, we have ph2 and ph4 above but not ph3 since it isn't used.

pranshuchittora replied:
Ideally we should keep all the values even if they aren't used.
This won't increase the bundle size as unwanted objects will be tree shaken during compile time

Jag96 replied:
Bundle size isn't the reason we want to avoid adding unused classes, the main reason is to avoid adding unnecessary code that can potentially be used by other contributors. In this case, we don't have a need for ph7, but if it is added in this PR then somebody could use ph7 later on and cause inconsistent styles across the app.

pranshuchittora replied:
Done

Jag96 replied:
Were these removed? I still see them in the project now, it looks like 7-10 are unused after the latest changes.

Jag96 reviewed (changes requested):
Conflicts and a few comments

pranshuchittora commented:
> Looks like the mobile view should have more padding around the inner logo/UI/etc.

![Screenshot from 2021-05-29 12-04-34](https://user-images.githubusercontent.com/32242596/120060565-18d1ab00-c076-11eb-8c1d-06bf28ec2ba4.png)

pranshuchittora committed:
feat: New landing page design

fix: Linting issues and removes unwanted props

fix: minor styling changes

fix: Terms of service component change for native

feat: PR review changes and removed unwatned props

Jag96 commented on src/styles/styles.js:
```diff
@@ -415,25 +438,43 @@ const styles = {
     },
 
+    signInPageWideLeftContainer: {
+        maxWidth: 400, minWidth: 350, width: '25%',
```
Please add line breaks for this

Jag96 commented on src/pages/signin/TermsAndLicenses/index.native.js:
```diff
@@ -1,3 +1,3 @@
+import TermsWithLicenses from './TermsWithLicenses';
 
-export default TermsOnly;
+export default TermsWithLicenses;
```
Since we were previously only showing terms on mobile but now we are showing terms with licenses, could you update the tests of this PR to ensure that QA knows that on mobile we should be showing licenses on mobile?

pranshuchittora replied:
Do you mean adding this to the **QA Steps** in the PR description?

Jag96 replied:
Yes, that works fine as well 👍. The Tests section is for PR reviewer tests and the QA section is for QA, but most of the time they're the same

Jag96 reviewed (commented):
Added a couple more comments, will let @shawnborton review the latest style

pranshuchittora committed:
chore: PR review changes

Jag96 reviewed (dismissed):
This is looking good to me, good work on this! A few things for @shawnborton to check on since I'm not sure if they're real issues:

1. Landscape iOS has a bunch of space at the top. Probably not an issue, but just double-checking w/ design
![image](https://user-images.githubusercontent.com/3981102/120530390-51cd9080-c392-11eb-9a8c-69c2764797e3.png)

2. Web with a small window has some tight spacing on the left and bottom, again just double-checking w/ design
![image](https://user-images.githubusercontent.com/3981102/120530505-7164b900-c392-11eb-9888-9279a8cbdd8a.png)

Will leave final review to @shawnborton

shawnborton commented:
Sorry for the delay - hoping to review this tomorrow.

pranshuchittora commented:
Hi, @shawnborton any update on this?

shawnborton commented:
Sorry for the delay! Looking into this now, going to start adding design comments. 

On desktop, the fine print does not have the same outer padding as the form above it:
![image](https://user-images.githubusercontent.com/2319350/120848117-3d6fcc00-c529-11eb-8b99-f341bc420f41.png)

shawnborton commented:
Do we need the padding on the bottom of `Forgot?`? Doing so makes a slight jump in the position of the input when you go from email to password:
![image](https://user-images.githubusercontent.com/2319350/120848309-8162d100-c529-11eb-8a2b-18c0bf556bc6.png)

shawnborton commented:
When the screen is in that weird middle size between mobile and desktop:
![image](https://user-images.githubusercontent.com/2319350/120848409-9c354580-c529-11eb-8208-0bd520953e53.png)


Let's actually make sure the content has a max-width and is just centered, like so:
![image](https://user-images.githubusercontent.com/2319350/120848601-d9013c80-c529-11eb-851e-d2c424a032d0.png)

shawnborton commented:
`Expensify` at the bottom of the sidebar should actually use a .svg and not be written in text:
![image](https://user-images.githubusercontent.com/2319350/120848761-0b129e80-c52a-11eb-83a9-81cd5c17941d.png)


You can use this file here and it should be shown at 78px wide: [expensify-wordmark.svg.zip](https://github.com/Expensify/Expensify.cash/files/6600647/expensify-wordmark.svg.zip)

shawnborton commented:
The sidebar itself should not be a fluid width on wide screens. I see you currently have it at 25%:
![image](https://user-images.githubusercontent.com/2319350/120849142-896f4080-c52a-11eb-9ec0-2ae55f425c1e.png)


It should actually always be fixed at 375, the same as the sidebar width when you are signed into the app. Once we hit the responsive breakpoint where we hide the blue area on the right, then the sidebar should just be 100% wide.

shawnborton commented:
Just chiming in as well to say I agree with Joe's feedback about this:

> Landscape iOS has a bunch of space at the top.

shawnborton commented:
Also just a heads up that I will be OOO next week, so tagging @michelle-thompson to help out with final design review while I am out.

pranshuchittora commented:
> Do we need the padding on the bottom of `Forgot?`? Doing so makes a slight jump in the position of the input when you go from email to password:
> ![image](https://user-images.githubusercontent.com/2319350/120848309-8162d100-c529-11eb-8a2b-18c0bf556bc6.png)

There is no padding 🤔

pranshuchittora commented:
Landscape Compatibility and max width 

https://user-images.githubusercontent.com/32242596/120931395-efb4b880-c70e-11eb-9b84-dfc1d450f2c4.mp4

pranshuchittora committed:
feat: PR review changes

stephanieelliott commented:
Hey @Jag96 I think this is blocked on your review? We have a tight deadline for this one, can you take a look when you have a chance?

michelle-thompson commented:
> There is no padding 🤔

As long as the input field doesn't move around when switching screens, that is what's important here.

And I think the landscape view on mobile looks good 👍

Jag96 reviewed (commented):
@pranshuchittora this is looking good, once conflicts are resolved I'll run through tests again

pranshuchittora committed:
Merge remote-tracking branch 'origin/main' into pranshuchittora/feat/landing-design

Jag96 commented on src/styles/utilities/spacing.js:
```diff
@@ -112,6 +112,14 @@ export default {
         marginTop: 24,
     },
 
+    mt25Percentage: {
```
It looks like this style is unused, if so can you please remove it?

Jag96 commented on src/styles/styles.js:
```diff
@@ -47,15 +47,37 @@ const styles = {
         fontSize: variables.fontSizeNormal,
+    },
+
+    h3Bold: {
```
It looks like this style is unused, if so please remove it.

Jag96 commented on src/languages/en.js:
```diff
@@ -70,7 +70,7 @@ export default {
     phoneCountryCode: '1',
     welcomeText: {
-        phrase1: 'With Expensify.cash, chat and payments are the same thing.',
+        phrase1: 'Welcome to Expensify.cash! Enter your phone or email to continue.',
```
Can you also please ensure that all of the new constants in this file are also added to `es.js`? They can remain in English, we just want to ensure any new keys exist in both files.

Jag96 commented on src/styles/styles.js:
```diff
@@ -47,15 +47,37 @@ const styles = {
         fontSize: variables.fontSizeLabel,
+    },
+
+    h4Bold: {
```
It looks like this style is unused, if so please remove it.

Jag96 commented on src/styles/styles.js:
```diff
@@ -47,15 +47,37 @@ const styles = {
         fontWeight: fontWeightBold,
     },
 
+    fontFamilyGTA: {
```
This style shouldn't be necessary, it is likely that it's being added to `Text` components that are included from `react-native` instead of `components/Text`. Could you update any places you're using this style to ensure the `Text` component is being imported from `components/Text` instead of `react-native`?

Jag96 reviewed (changes requested):
It looks like the homepage text links are the same color as the background, which makes it hard to read:

![image](https://user-images.githubusercontent.com/3981102/121438025-21f52e80-c938-11eb-98eb-44ee3a3bd9d0.png)
![image](https://user-images.githubusercontent.com/3981102/121438033-2588b580-c938-11eb-89d7-5f1a9ca9fa5c.png)

Can we update those to be readable like they are in the video description?

Also, the links on Android are a bit large:
![image](https://user-images.githubusercontent.com/3981102/121438406-d000d880-c938-11eb-8c3c-661ec21afc97.png)

pranshuchittora commented:
Hi @Jag96 the TextLink is not working properly when used inline. Can we get the previous API back the open URL thing

Jag96 commented:
> the TextLink is not working properly when used inline
Is it just the styling that is not working? Or is there other functionality?

> Can we get the previous API back the open URL thing
I'm not sure what previous API you're referring to, could you reference a previous commit to make it easier to understand what previous functionality you're referring to? Also, if this issue affects all inline TextLinks, our [slack channel](https://expensify.slack.com/archives/C01GTK53T8Q) would be a good place to bring that up

pranshuchittora commented:
> > the TextLink is not working properly when used inline
> 
> Is it just the styling that is not working? Or is there other functionality?
> 
> > Can we get the previous API back the open URL thing
> 
> I'm not sure what previous API you're referring to, could you reference a previous commit to make it easier to understand what previous functionality you're referring to? Also, if this issue affects all inline TextLinks, our [slack channel](https://expensify.slack.com/archives/C01GTK53T8Q) would be a good place to bring that up

I am talking about this -> https://github.com/Expensify/Expensify.cash/tree/c97a0d1af7c706ae916dbcb8a05f30b529978360/src/libs/openURLInNewTab

pranshuchittora committed:
feat: Added open in new tab

pranshuchittora committed:
Merge remote-tracking branch 'origin/main' into pranshuchittora/feat/landing-design

pranshuchittora commented:
@Jag96 I have updated the PR

Jag96 reviewed (changes requested):
It looks like in [this commit](https://github.com/Expensify/Expensify.cash/pull/3079/commits/71ae6cbf13efc2031adbf385d92798741d33a0a9) you re-added the `openURLInNewTab` functionality that was removed in https://github.com/Expensify/Expensify.cash/pull/3311. I'd rather not re-introduce that functionality if there is another way to fix the issue.

It looks like the regression is mentioned in https://github.com/Expensify/Expensify.cash/issues/3100#issuecomment-857242234 and the solution is implemented in https://github.com/Expensify/Expensify.cash/pull/3502. Will that solution work for this?

pranshuchittora commented:
New text link api is not inline friendly. 

Just open the https://expensify.cash/ and make the screen smaller, you can see that the full stop after privacy policy goes to the new line as it is part of the next sentence. I think someone missed this while reviewing.
![Screenshot from 2021-06-12 01-25-58](https://user-images.githubusercontent.com/32242596/121742241-321a3300-cb1d-11eb-94a2-18703540bb63.png)

tugbadogan commented:
@pranshuchittora #3502 solves this issue, but it is not shipped to production yet. You can confirm it's working by loading the staging server via https://staging.expensify.cash

pranshuchittora commented:
> @pranshuchittora #3502 solves this issue, but it is not shipped to production yet. You can confirm it's working by loading the staging server via https://staging.expensify.cash

I try migrating again but experiencing some weird behaviour. 

https://user-images.githubusercontent.com/32242596/121744178-25e3a500-cb20-11eb-906c-14c8e26e07f6.mp4

```jsx
import React from 'react';
import {Text, View} from 'react-native';
import styles from '../../../styles/styles';
import CONST from '../../../CONST';
import TextLink from '../../../components/TextLink';
import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
import LogoWordmark from '../../../../assets/images/expensify-wordmark.svg';

const TermsWithLicenses = ({translate}) => (
    <View>
        <View style={[styles.mt1, styles.alignItemsCenter]}>
            <LogoWordmark height={30} width={80} />
        </View>
        <View style={[styles.mt2, styles.flexRow, styles.flexWrap, styles.textAlignCenter]}>
            <Text style={[styles.loginTermsText]}>
                {translate('termsOfUse.phrase1')}
                {' '}
            </Text>
            <TextLink style={[styles.loginTermsText]} href={CONST.TERMS_URL}>
                {translate('termsOfUse.phrase2')}
            </TextLink>
            <Text style={[styles.loginTermsText]}>
                {' '}
                {translate('termsOfUse.phrase3')}
                {' '}
            </Text>
            <TextLink style={[styles.loginTermsText]} href={CONST.PRIVACY_URL}>
                {translate('termsOfUse.phrase4')}
            </TextLink>
            <Text style={[styles.loginTermsText]}>.</Text>
            <Text>
                <Text style={[styles.loginTermsText]}>
                    {translate('termsOfUse.phrase5')}
                    {' '}
                </Text>
                <TextLink style={[styles.loginTermsText]} href={CONST.LICENSES_URL}>
                    {translate('termsOfUse.phrase6')}
                </TextLink>
                <Text style={[styles.loginTermsText]}>.</Text>
            </Text>
        </View>
    </View>
);

TermsWithLicenses.propTypes = {...withLocalizePropTypes};

export default withLocalize(TermsWithLicenses);

```

pranshuchittora committed:
feat: Migration to text link

Jag96 commented:
> I have tried migrating to TextLink again but experiencing some weird behaviour (Terms are being center aligned but the behaviour is not consistent). Can you help resolving them

@pranshuchittora could you elaborate on what the weird behavior is? Is it that the `Money transmission...` text is not aligning with the `By logging in` text? Or is it still `the full stop after privacy policy goes to the new line as it is part of the next sentence`?

Neither of these issues appear on main/staging, so they are related to this branch, and they both look like CSS issues. 

If the only issue is that when making the web window smaller, it is possible for the `.` in `privacy policy.` to be on a new line, I don't think that is worth blocking this PR since it only happens if the window is 291 or 292 pixels wide (just those values). Also, on staging, the `privacy policy.` text is always on the next line, so this issue could still exist on previous branches, it likely just showed itself now since you've changed `max-width` for some components as part of these updates.

pranshuchittora commented:
> > I have tried migrating to TextLink again but experiencing some weird behaviour (Terms are being center aligned but the behaviour is not consistent). Can you help resolving them
> 
> @pranshuchittora could you elaborate on what the weird behavior is? Is it that the `Money transmission...` text is not aligning with the `By logging in` text? Or is it still `the full stop after privacy policy goes to the new line as it is part of the next sentence`?
> 
> Neither of these issues appear on main/staging, so they are related to this branch, and they both look like CSS issues.
> 
> If the only issue is that when making the web window smaller, it is possible for the `.` in `privacy policy.` to be on a new line, I don't think that is worth blocking this PR since it only happens if the window is 291 or 292 pixels wide (just those values). Also, on staging, the `privacy policy.` text is always on the next line, so this issue could still exist on previous branches, it likely just showed itself now since you've changed `max-width` for some components as part of these updates.

I agree that it is not a blocker but using text link with center aligning content is behaving weirdly. This won’t appear in staging as it is not center aligned/justified in the current build. 

No issue I will update the PR with `TextLink` component.

pranshuchittora committed:
feat: Migrated to TextLink

pranshuchittora commented:
@Jag96 I have updated the PR, it's ready to merge

Jag96 commented on src/pages/signin/TermsAndLicenses/TermsWithLicenses.js:
```diff
@@ -4,35 +4,41 @@ import styles from '../../../styles/styles';
-        </Text>
+            <Text>
+                <Text style={[styles.loginTermsText]}>
+                    {translate('termsOfUse.phrase5')}
```
It looks like there should be a `{' '}` above this as well since there is no space between `privacy policy.` and `Money transmission`. Additionally, `licences` is not aligned properly in on iOS.

<img src='https://user-images.githubusercontent.com/3981102/122138110-edcbb300-cdfa-11eb-9346-e0fa196b6696.png' width='400'/>

pranshuchittora replied:
Flex wrap is behaving weirdly in mobile. I tried fixing this but due to the long content size it's getting messed up. Can you pls help me with fixing this.

![Screenshot from 2021-06-16 06-56-16](https://user-images.githubusercontent.com/32242596/122143913-0aee9900-ce70-11eb-900f-e0fd32e26687.png)

![Screenshot from 2021-06-16 06-56-31](https://user-images.githubusercontent.com/32242596/122143910-09bd6c00-ce70-11eb-8cc8-4da168785995.png)

```jsx
import React from 'react';
import {Text, View} from 'react-native';
import styles from '../../../styles/styles';
import CONST from '../../../CONST';
import TextLink from '../../../components/TextLink';
import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
import LogoWordmark from '../../../../assets/images/expensify-wordmark.svg';

const TermsWithLicenses = ({translate}) => (
    <View>
        <View style={[styles.mt1, styles.alignItemsCenter]}>
            <LogoWordmark height={30} width={80} />
        </View>
        <View style={[styles.mt2, styles.flexRow, styles.flexWrap, styles.textAlignCenter]}>
            <Text style={[styles.loginTermsText]}>
                {translate('termsOfUse.phrase1')}
                {' '}
            </Text>
            <TextLink style={[styles.loginTermsText]} href={CONST.TERMS_URL}>
                {translate('termsOfUse.phrase2')}
            </TextLink>
            <Text style={[styles.loginTermsText]}>
                {' '}
                {translate('termsOfUse.phrase3')}
                {' '}
            </Text>
            <TextLink style={[styles.loginTermsText]} href={CONST.PRIVACY_URL}>
                {translate('termsOfUse.phrase4')}
            </TextLink>
            <Text style={[styles.loginTermsText]}>.</Text>
            <Text style={[styles.loginTermsText]}>
                {translate('termsOfUse.phrase5')}
                {' '}
            </Text>
            <TextLink style={[styles.loginTermsText]} href={CONST.LICENSES_URL}>
                {translate('termsOfUse.phrase6')}
            </TextLink>
            <Text style={[styles.loginTermsText]}>.</Text>
        </View>
    </View>
);

TermsWithLicenses.propTypes = {...withLocalizePropTypes};

export default withLocalize(TermsWithLicenses);

```

Jag96 replied:
> I tried fixing this but due to the long content size it's getting messed up. Can you pls help me with fixing this.

If you are having trouble with a solution please post in our [public slack channel](https://expensify.slack.com/archives/C01GTK53T8Q) and ask for help. I'm not sure what the issue is here so it is more likely that our community can help you faster than I can alone.

pranshuchittora replied:
Fixed

Jag96 reviewed (changes requested):
Found 2 UI issues on iOS

Jag96 commented:
Requesting a review from Pullerbear to help with reviews since I'll be OOO the rest of the week

pranshuchittora commented:
> Requesting a review from Pullerbear to help with reviews since I'll be OOO the rest of the week

Headup, In the current main branch `TermsWithLicenses` is also behaving similarly (i.e. have aligning issues) in native. 
https://expensify.slack.com/archives/C01GTK53T8Q/p1623814030375300
 
![Screenshot from 2021-06-16 08-52-37](https://user-images.githubusercontent.com/32242596/122153556-e18a3900-ce80-11eb-936d-41602dcc2451.png)

Beamanator reviewed (changes requested):
One minor spelling fix + I believe the designs show both license phrases centered horizontally - this is how I solved that & the licenses floating up issues:

Centering the phrases:

```js
const TermsWithLicenses = ({translate}) => (
    <View>
        <View style={[styles.mt1, styles.alignItemsCenter]}>
            <LogoWordmark height={30} width={80} />
        </View>
        <View>
            <Text style={[styles.flexColumn, styles.textAlignCenter]}>
                <Text style={[styles.loginTermsText]}>
                    {translate('termsOfUse.phrase1')}
                    {' '}
                </Text>
                <TextLink style={[styles.loginTermsText]} href={CONST.TERMS_URL}>
                    {translate('termsOfUse.phrase2')}
                </TextLink>
                <Text style={[styles.loginTermsText]}>
                    {' '}
                    {translate('termsOfUse.phrase3')}
                    {' '}
                </Text>
                <TextLink style={[styles.loginTermsText]} href={CONST.PRIVACY_URL}>
                    {translate('termsOfUse.phrase4')}
                </TextLink>
                <Text style={[styles.loginTermsText]}>.</Text>
            </Text>
        </View>
        <View>
            <Text style={[styles.flexColumn, styles.textAlignCenter]}>
                <Text style={[styles.loginTermsText]}>
                    {translate('termsOfUse.phrase5')}
                    {' '}
                </Text>
                <TextLink style={[styles.loginTermsText]} href={CONST.LICENSES_URL}>
                    {translate('termsOfUse.phrase6')}
                </TextLink>
                <Text style={[styles.loginTermsText]}>.</Text>
            </Text>
        </View>
    </View>
);
```

Fixing the links floating up:

```diff
const TextLink = (props) => {
    const additionalStyles = _.isArray(props.style) ? props.style : [props.style];
    return (
        <Pressable
            onPress={(e) => {
                e.preventDefault();
                Linking.openURL(props.href);
            }}
            accessibilityRole="link"
            href={props.href}
+           style={{margin: -1}}
        >
            {({hovered, pressed}) => (
                <Text style={[...additionalStyles, styles.link, (hovered || pressed) ? styles.linkHovered : undefined]}>
                    {props.children}
                </Text>
            )}
        </Pressable>
    );
};
```

I believe the above vertical-spacing fix works because `TextLink` wraps a `Pressable`, which wraps a `View` which has a bit of styling on it by default - including some margins or padding which can get removed with `margin: -1`

Feel free to use those solutions or modify them if any further changes are needed :)

pranshuchittora commented:
Links are still floating 

![Screenshot from 2021-06-17 18-53-53](https://user-images.githubusercontent.com/32242596/122405228-7f732600-cf9d-11eb-804c-0e65ed74dc2b.png)

pranshuchittora commented:
Links are still floating 

![Screenshot from 2021-06-17 18-53-53](https://user-images.githubusercontent.com/32242596/122405228-7f732600-cf9d-11eb-804c-0e65ed74dc2b.png)

Able to make it work via 
```jsx
  <TextLink style={[styles.loginTermsText, {marginVertical: -4}]} href={CONST.LICENSES_URL}>
                    {translate('termsOfUse.phrase6')}
  </TextLink>
```
![Screenshot from 2021-06-17 19-05-15](https://user-images.githubusercontent.com/32242596/122407220-168cad80-cf9f-11eb-8c0e-fd86bc00e30b.png)


Is that the correct approach @Beamanator

pranshuchittora committed:
fix: Floating links

pranshuchittora commented:
All the issues have been fixed :)

pranshuchittora committed:
fix: Removed inline styles

pranshuchittora committed:
Merge branch 'main' into pranshuchittora/feat/landing-design

shawnborton commented:
Are all of your screenshots in the original comment up to date? If so, then I wanted to confirm that you saw my earlier feedback that the mobile view of the sign in form should have the form controls centered with a max-width of 295 like so:

![image](https://user-images.githubusercontent.com/2319350/122439365-7314ac00-cf50-11eb-81e5-dc499f0f044f.png)

pranshuchittora committed:
fix: Styling changes

pranshuchittora commented:
I will be updating the screenshots. 

Apart from that 295 will look very compact. Here's the 295 width
![Screenshot from 2021-06-17 22-18-35](https://user-images.githubusercontent.com/32242596/122440257-0209de80-cfba-11eb-856b-9c215598c199.png)

shawnborton commented:
Sorry for the confusion, but I am suggesting that the form contents themselves should be at 295.

pranshuchittora committed:
feat: PR review changes

pranshuchittora commented:
@stephanieelliott this PR is ready for final review -> https://expensify.slack.com/archives/C01GTK53T8Q/p1623949003407200

pranshuchittora committed:
fix: Styling fixes

pranshuchittora committed:
chore: Removed unwanted files

pranshuchittora committed:
fix: Minor styling changes

Jag96 commented on src/styles/utilities/spacing.js:
```diff
@@ -184,6 +188,10 @@ export default {
         paddingVertical: 20,
     },
 
+    pv8: {
```
Is this used? If not, please remove it

Jag96 reviewed (changes requested):
I noticed a couple of things while testing:

1. On Web, there is an error in the console when the sign in page loads:
![image](https://user-images.githubusercontent.com/3981102/122848636-cf642c80-d2be-11eb-902d-00c80fd68b80.png)
2. On iOS, the text links are still not completely lined up (see `licenses`):
<img src='https://user-images.githubusercontent.com/3981102/122848651-dc811b80-d2be-11eb-9d48-904081ec23e4.png' width='400'/>

pranshuchittora committed:
fix: Terms and licenses fix

pranshuchittora committed:
fix: Textlink

pranshuchittora committed:
fix: Removed unused styles

pranshuchittora committed:
Merge remote-tracking branch 'origin/main' into pranshuchittora/feat/landing-design

pranshuchittora committed:
fix: Linting fixes

Jag96 commented on src/pages/signin/TermsAndLicenses/TermsWithLicenses.js:
```diff
@@ -3,35 +3,49 @@ import {Text, View} from 'react-native';
+            {translate('termsOfUse.phrase5')}
+            {' '}
+
+
```
Remove unnecessary spaces here

pranshuchittora replied:
In android, the behaviour is a bit different therefore I have used Platform.select

roryabraham replied:
Please don't use `Platform.select` - use [platform-specific file extensions](https://github.com/Expensify/Expensify.cash#platform-specific-file-extensions) instead

Jag96 reviewed (commented):
It still looks like the terms on iOS have inconsistent spacing for TextLinks:
![image](https://user-images.githubusercontent.com/3981102/123017855-4d3d3c00-d382-11eb-8388-a7c5b4310d2d.png)

pranshuchittora committed:
fix: Minor changes to the terms with licenses

pranshuchittora committed:
fix: Platform patches

pranshuchittora committed:
fix: Removed unwanted spaces

pranshuchittora committed:
feat: Terns with Licenses for ios

pranshuchittora commented:
I have updated the PR

iOS
<img width="291" alt="Screenshot 2021-06-24 at 6 15 12 PM" src="https://user-images.githubusercontent.com/32242596/123266329-85c34e00-d519-11eb-935c-22b12abc85fa.png">

Jag96 reviewed (dismissed):
LGTM, @Beamanator all yours

Beamanator reviewed (changes requested):
A few comments, mainly about fixing the way we should handle platform-specific files.

One more thing: I see that the android & ios versions of `TermsWithLicenses` are almost identical, except for the addition of `styles.termsLinkIos` on iOS (and a `termsOfUse.phrase6` and `.phrase7` are grouped within a `View` tag for iOS, not for android). If we can make those components exactly the same (except for the style change for ios), we can just make that margin style adjustment a utility (in `/styles/utilities`) and make `marginTop: -1` platform-specific (in a `index.ios.js` file). This way, we can have 1 component for both native platforms, making the `TermsWithLicenses` component(s) easier to maintain. Let me know your thoughts!

pranshuchittora committed:
feat: Minor file structure changes

pranshuchittora commented:
@Beamanator I have updated the PR.
1. I have updated the file structure.
2. For H3 styles I have removed the font-weight property and added a separate style named `textBold` with this approach we can have both `H3` and `H3 Bold` this is more maintainable in the long run.
3. Regarding the merging of the ios and android, For the past 3 weeks I have been trying to fix the TermsWithLicenses weird behaviour in different platforms. Once again I tried to merge the 2 but no success.

Beamanator reviewed (dismissed):
Nice restructure of the platform-specific files 👍 

Thanks for trying to combine the `TermsWithLicenses` for ios & android - no problem that they couldn't be combined :)

@Jag96 back to you, and I think it would be nice to get a final approval from @shawnborton

shawnborton commented:
Everything looks pretty good to me, but after seeing some screenshots on Web/Desktop, I think we might want to use some kind of fixed outer padding/gutter around the main content inside the blue area. From Figma, you can see where we have a fixed 80px on each side of the inner content in the blue area:
![image](https://user-images.githubusercontent.com/2319350/124519347-e23a2f00-dd9d-11eb-903d-f8c44e645d1c.png)

From the screenshots, you can see where that outer padding gets very skinny:
![image](https://user-images.githubusercontent.com/2319350/124519377-f2520e80-dd9d-11eb-82c9-c328e8d0a359.png)


Thoughts on that?

Beamanator commented:
I like that idea @shawnborton ! 👍

Jag96 reviewed (changes requested):
Added one request in addition to the design request above

pranshuchittora committed:
fix: Minor styling changes

pranshuchittora commented on src/pages/EnablePayments/OnfidoStep.js:
```diff
@@ -120,7 +120,7 @@ class OnfidoStep extends React.Component {
                             {!this.props.walletOnfidoData.loading && this.props.walletOnfidoData.error && (
                                 <>
-                                    <Text style={[styles.h3, styles.mb2]}>
+                                    <Text style={[styles.h3, styles.textBold, styles.mb2]}>
```
There's already a style names `testStrong`

https://github.com/Expensify/Expensify.cash/blob/d3b3a797584054364d96effd00b746bcca77da82/src/styles/styles.js#L96

This has fontFamily that is the reason for `textBold`. Not sure whether we can use this instead of `textBold` (for all use cases)

Jag96 replied:
I'm recommending we use the existing class `textStrong` here and remove the new `textBold` class. There are currently no usages of `fontWeight: fontWeightBold,` without `fontFamily: fontFamily.GTA_BOLD,` so I think we should keep it consistent.

Jag96 commented on src/styles/styles.js:
```diff
@@ -47,13 +47,19 @@ const styles = {
     h3: {
         fontFamily: fontFamily.GTA_BOLD,
         fontSize: variables.fontSizeNormal,
-        fontWeight: fontWeightBold,
```
Any reason this was removed here? Doesn't this affect other h3/h4 elements?

pranshuchittora commented:
> Any reason this was removed here? Doesn't this affect other h3/h4 elements? If you add this back you wouldn't need styles.textBold or styles.textStrong

So the use case is to use non bold version of h3

pranshuchittora committed:
fix: Minor style name changes

Jag96 reviewed (dismissed):
LGTM, will leave final approval/merge to @shawnborton

shawnborton commented on src/pages/signin/SignInPageLayout/SignInPageLayoutWide.js:
```diff
@@ -1,92 +1,104 @@
+                    styles.flex1,
+                    styles.dFlex,
+                    styles.flexColumn,
+                    styles.mt40Percentage,
```
Instead of using 40% as the margin top, what if we did something that was actually responsive to the viewport - so something like `15vh`? Does React Native work well with `vh` units?

pranshuchittora replied:
No react-native doesn't work with `vh`. Though a workaround is possible but might require more effort.
`%` somewhat works in a similar fashion

shawnborton replied:
Hmm it didn't seem like the margin was actually being responsive though, it felt fixed when resizing the page. Any ideas?

pranshuchittora replied:
To solve this we need to apply different styles based on the platform. Won't be maintainable in the long run.

shawnborton replied:
Can you elaborate on what you mean? The left area of the sign up page is basically the same view used on desktop or mobile. The only difference is that on desktop, there is a blue area added to the right of it.

shawnborton commented:
We're super close! I left a comment about the margin on top of the form in the left side. My other comment is around the phone image on the blue background. I feel like we don't need to make it react so much to the viewport height. Basically I think it can have a fixed height and a max-height at the same time. Maybe said another way, the phone image should generally always be the same size but in extreme circumstances (if the viewport height is super short), it can shrink down. But in the other case, like when the viewport height gets tall, I think it should have a max-height and not keep getting larger. Otherwise we would get this:
![image](https://user-images.githubusercontent.com/2319350/125003375-ef5c5580-e00b-11eb-9120-c8e6fd641f10.png)


In that scenario, I think it would be better if the phone stayed the same size in relationship to the text to the left. We would then just take the text + phone and vertically center up that content in the blue area. 

Here is a quick video of me resizing my browser so you can see it in action:
![o7hYl0Vslh](https://user-images.githubusercontent.com/2319350/125003486-2894c580-e00c-11eb-8385-f602e4601967.gif)

Let me know if that makes sense - I'm happy to mock something up if needed.

MitchExpensify commented:
Can we also switch out all Expensify.cash refernces to "New.Expensify.com" please?

pranshuchittora commented:
> My other comment is around the phone image on the blue background. I feel like we don't need to make it react so much to the viewport height. Basically I think it can have a fixed height and a max-height at the same time. Maybe said another way, the phone image should generally always be the same size but in extreme circumstances (if the viewport height is super short), it can shrink down. But in the other case, like when the viewport height gets tall, I think it should have a max-height and not keep getting larger. Otherwise we would get this:

Making the height fixed will introduce a scrollbar which is not a good approach to responsiveness.

shawnborton commented:
> Making the height fixed will introduce a scrollbar which is not a good approach to responsiveness.

I think we need to balance reasonability with responsiveness though. We don't need the app to responsively scale to extreme, non-common height. If we can have a fixed image that works when the app is at say 600-700px tall, then I think we safely optimize for a majority of users.

pranshuchittora commented:
Ok then what should be the plan of action

kevinksullivan commented:
Just want to bump [the change](https://github.com/Expensify/Expensify.cash/pull/3079#issuecomment-877284151) to use "New Expensify" in place of "Expensify.cash" in the two areas mentioned. Can we incorporate this into the PR?

shawnborton commented:
I think the path forward is pretty simple with these small adjustments: 
![image](https://user-images.githubusercontent.com/2319350/125356916-bda10280-e31b-11eb-9d40-0d3254062a1b.png)

shawnborton commented on src/styles/styles.js:
```diff
@@ -420,22 +433,20 @@ const styles = {
-        width: 295,
+    signInWelcomeScreenshotWide: {
+        aspectRatio: 1,
+        width: '80%',
```
Let's change this to 100%

shawnborton commented on src/styles/styles.js:
```diff
@@ -420,22 +433,20 @@ const styles = {
+    signInWelcomeScreenshotWide: {
+        aspectRatio: 1,
+        width: '80%',
+        height: '80%',
```
Let's change this to 100%

shawnborton commented on src/styles/styles.js:
```diff
@@ -420,22 +433,20 @@ const styles = {
+        aspectRatio: 1,
+        width: '80%',
+        height: '80%',
+        maxHeight: 1000,
```
Let's change this to be the size of the asset, so it should be 551.

pranshuchittora committed:
fix: Minor changes

shawnborton commented:
Looks good but:
- Can you please address [the comment here](https://github.com/Expensify/Expensify.cash/pull/3079#issuecomment-878359072)?
- Address conflicts

pranshuchittora committed:
Merge remote-tracking branch 'origin/main' into pranshuchittora/feat/landing-design

pranshuchittora commented:
> Looks good but:
> 
> * Can you please address [the comment here](https://github.com/Expensify/Expensify.cash/pull/3079#issuecomment-878359072)?
> * Address conflicts

Sure

pranshuchittora committed:
fix: Minor string changes

pranshuchittora committed:
chore: Minor string changes

pranshuchittora commented:
@shawnborton I have updated the PR with the final changes.

<img width="1435" alt="Screenshot 2021-07-14 at 12 21 05 AM" src="https://user-images.githubusercontent.com/32242596/125508649-05718c81-dce5-4db3-8f03-4e81ede2ff0f.png">

shawnborton commented:
Thanks for the update! Do you know what's happening with the fine print here?
![image](https://user-images.githubusercontent.com/2319350/125508960-d19ad4eb-db17-4546-91a6-c00e2fe16a2c.png)


And cc @kevinksullivan - should it be "New Expensify" or "new Expensify"?

pranshuchittora committed:
fix: Minor string changes

shawnborton commented:
@pranshuchittora it also looks like you missed changing out `Expensify.cash` here:
![image](https://user-images.githubusercontent.com/2319350/125509039-ed67097a-4f85-4956-9316-e96ab12ddcbe.png)

pranshuchittora commented:
Done

<img width="1440" alt="Screenshot 2021-07-14 at 12 24 33 AM" src="https://user-images.githubusercontent.com/32242596/125509079-15fbd38a-074b-4282-a5a8-19a28c62fbd5.png">

pranshuchittora commented:
Wait somebody again changes the TermsWithLicenses 🤦🏼

pranshuchittora committed:
fix: File refs

pranshuchittora commented:
Fixed, it was a file reference issue that occurred due to rebasing with the main.


![Simulator Screen Shot - iPhone 12 - 2021-07-14 at 01 05 55](https://user-images.githubusercontent.com/32242596/125514362-154ba058-3999-4a6c-b9af-7fb7617e7b7f.png)

pranshuchittora committed:
fix: Minor terms styling changes

joelbettner commented on src/languages/en.js:
```diff
@@ -259,17 +259,26 @@ export default {
     signInPage: {
         expensifyDotCash: 'Expensify.cash',
-        expensifyIsOpenSource: 'Expensify.cash is open source',
+        expensifyIsOpenSource: 'New Expensify is open source',
```
@kevinksullivan is this how we want it to read? As stated elsewhere, this just seems clunky/grammatically incorrect to me.

joelbettner replied:
Perhaps `"The new Expensify is open source"`?

joelbettner commented on src/languages/en.js:
```diff
@@ -95,7 +95,7 @@ export default {
     phoneCountryCode: '1',
     welcomeText: {
-        phrase1: 'With Expensify.cash, chat and payments are the same thing.',
+        phrase1: 'Welcome to new Expensify! Enter your phone or email to continue.',
```
This language is confusing. Perhaps `'Welcome to the new Expensify! Enter your phone number or email to continue.'`

@kevinksullivan does it not seem weird to you to have it say `"Welcome to new Expensify!"`?

kevinksullivan replied:
I agree. I think `the New Expensify` is best as the latest mockup shows.

joelbettner commented on src/languages/en.js:
```diff
@@ -259,17 +259,26 @@ export default {
-        phrase5: 'Money transmission is provided by Expensify Payments LLC (NMLS ID:2017010) pursuant to its',
-        phrase6: 'licenses',
+        phrase5: 'Money transmission is provided by Expensify Payments LLC (NMLS',
+        phrase6: 'ID:2017010) pursuant to its',
```
Not sure why `phrase5` and `phrase6` are separated. They are both handled the same way:

```js
            <Text style={[styles.textAlignCenter, styles.loginTermsText]}>
                {translate('termsOfUse.phrase5')}
                {' '}
            </Text>
            <Text style={[styles.textAlignCenter, styles.loginTermsText]}>
                {translate('termsOfUse.phrase6')}
            </Text>
```

Could we just consolidate them down into one phrase line?

pranshuchittora replied:
No, we can't I have discussed this earlier. The expected behaviour is that they both come on different lines.

pranshuchittora committed:
fix: adhoc string changes

pranshuchittora commented:
@joelbettner I have updated the PR. cc @shawnborton 

<img width="1440" alt="Screenshot 2021-07-14 at 2 21 56 AM" src="https://user-images.githubusercontent.com/32242596/125523294-170c90eb-731e-4625-9d19-0c9fa6e231a5.png">

shawnborton commented:
@pranshuchittora Thank you - one last change - can you please capitalize the "N" in `New Expensify`?

pranshuchittora committed:
fix: Minor string changes

pranshuchittora commented:
Done @shawnborton 

<img width="1440" alt="Screenshot 2021-07-14 at 2 28 33 AM" src="https://user-images.githubusercontent.com/32242596/125524073-930cfcf7-b5c6-4c62-b2fc-862079cea181.png">

Jag96 reviewed (approved):
LGTM!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/Expensify.cash/actions/runs/1031434935) to staging         in version: 1.0.77-6🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

isagoico commented:
## Web - Profile -  Too much space in mweb safari in log in page

## Action Performed
1. Go to staging.expensify.cash
## Expected Result
Log in page should be displayed properly
## Actual Result
Too much space in mweb safari in log in page

**Build:** 1.0.78-0

**Notes/Images/Video**
OS and browser used: Safari

<img src="https://user-images.githubusercontent.com/44479856/125796925-349d7dae-0864-426e-b921-de82ce82feba.png" width="40%">

pranshuchittora commented:
This is the expected behaviour

![image](https://user-images.githubusercontent.com/32242596/125803502-9365d654-f316-4315-a8e3-6501b969d50d.png)

shawnborton commented:
True, but I do think it would be nice if that top margin above the icon used a relative unit. Remember I left that comment here: https://github.com/Expensify/Expensify.cash/pull/3079#discussion_r666581784

We have this in place on our Expensify.com homepage:
![image](https://user-images.githubusercontent.com/2319350/126206920-91b1d6ee-550c-458f-a528-d4ac526f9db4.png)

Where the top padding is `15vh`. Perhaps we can create a follow up issue to make this a bit more mobile friendly. To clarify, I don't think this is a blocker for paying @pranshuchittora for the work that was completed, I just think we can separately follow up with some enhancements.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/1060984705) to production         in version: 1.0.79-4🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
